kin/client.py

```diff
@@ -0,0 +1,370 @@
+                balances['XLM'] = balance.balance
+
+        return balances
+
+    def get_account_status(self, address):
+        """
+        Get a given account status
+        :param str address: The public address of the account to query.
+        :return: One the possible account statuses.
+        :rtype: :enum:`kin.AccountStatus`
+        """
+        try:
+            balances = self.get_account_balances(address)
+        except KinErrors.AccountNotFoundError:
+            return AccountStatus.NOT_CREATED
+
+        try:
+            balances['KIN']
+        except KeyError:
+            return AccountStatus.NOT_ACTIVATED
+
+        return AccountStatus.ACTIVATED
+
+    def get_account_data(self, address):
+        """Get account data.
+
+        :param str address: the public address of the account to query.
+
+        :return: account data
+        :rtype: :class:`kin.AccountData`
+
+        :raises: ValueError: if the provided address has a wrong format.
+        :raises: :class:`KinErrors.AccountNotFoundError`: if the account does not exist.
+        """
+        # TODO: might want to simplify the returning data
+        if not is_valid_address(address):
+            raise ValueError('invalid address: {}'.format(address))
+
+        try:
+            acc = self.horizon.account(address)
+            return AccountData(acc, strict=False)
+        except Exception as e:
+            err = KinErrors.translate_error(e)
+            raise KinErrors.AccountNotFoundError(address) if \
+                isinstance(err, KinErrors.ResourceNotFoundError) else err
+
+    def get_transaction_data(self, tx_hash, simple=True):
+        """Gets transaction data.
+
+        :param str tx_hash: transaction hash.
+        :param boolean simple: (optional) returns a simplified transaction object
+
+        :return: transaction data
+        :rtype: :class:`kin.TransactionData` or `kin.SimplifiedTransaction`
+
+        :raises: ValueError: if the provided hash is invalid.
+        :raises: :class:`KinErrors.ResourceNotFoundError`: if the transaction does not exist.
+        :raises: :class:`KinErrors.CantSimplifyError`: if the tx is too complex to simplify
+        """
+        if not is_valid_transaction_hash(tx_hash):
+            raise ValueError('invalid transaction hash: {}'.format(tx_hash))
+
+        try:
+            tx = self.horizon.transaction(tx_hash)
+
+            # get transaction operations
+            tx_ops = self.horizon.transaction_operations(tx['hash'], params={'limit': 100})
+            tx['operations'] = tx_ops['_embedded']['records']
+
+            tx_data = TransactionData(tx, strict=False)
+        except Exception as e:
+            raise KinErrors.translate_error(e)
+
+        if simple:
+            return SimplifiedTransaction(tx_data, self.kin_asset)
+        return tx_data
+
+    def verify_kin_payment(self, tx_hash, source, destination, amount, memo=None, check_memo=False):
```

Comment:
`check_memo` seems redundant, if memo is specified then it should be checked o.w. no

Reply:
:+1: